apps/src/templates/projects/ProjectCardGrid.jsx

```diff
@@ -0,0 +1,155 @@
+        playlabProjects.push(project);
+      }
+      if (project.projectData.type === 'weblab') {
+        weblabProjects.push(project);
+      }
+    });
+
+    let projectsSortedByType = {
+      applabProjects,
+      gamelabProjects,
+      artistProjects,
+      playlabProjects,
+      weblabProjects
+    };
+    return projectsSortedByType;
+  },
+
+//write a render helper function that renders the correct projects based on gallery type
+
+  renderSpecificGallery() {
+    const { projects, galleryType } = this.props;
+
+    if (galleryType === 'public'){
+
+      return (
+        <div style={styles.grid}>
+          <h2 style={styles.labHeading}> App Lab </h2>
+          {this.sortByType(projects).applabProjects.slice(0,4).map((project, index) => (
+            <div key={index} style={styles.card}>
+              <ProjectCard
+                projectData={project.projectData}
+                currentGallery={galleryType}
+              />
+            </div>
+          ))}
+          <h2 style={styles.labHeading}> Game Lab </h2>
+          {this.sortByType(projects).gamelabProjects.slice(0,4).map((project, index) => (
+            <div key={index} style={styles.card}>
+              <ProjectCard
+                projectData={project.projectData}
+                currentGallery={galleryType}
+              />
+            </div>
+          ))}
+          <h2 style={styles.labHeading}> Artist </h2>
+          {this.sortByType(projects).artistProjects.slice(0,4).map((project, index) => (
+            <div key={index} style={styles.card}>
+              <ProjectCard
+                projectData={project.projectData}
+                currentGallery={galleryType}
+              />
+            </div>
+          ))}
+          <h2 style={styles.labHeading}> Play Lab </h2>
+          {this.sortByType(projects).playlabProjects.slice(0,4).map((project, index) => (
+            <div key={index} style={styles.card}>
+              <ProjectCard
+                projectData={project.projectData}
+                currentGallery={galleryType}
+              />
+            </div>
+          ))}
+          <h2 style={styles.labHeading}> Web Lab </h2>
+          {this.sortByType(projects).weblabProjects.slice(0,4).map((project, index) => (
+            <div key={index} style={styles.card}>
+              <ProjectCard
+                projectData={project.projectData}
+                currentGallery={galleryType}
+              />
+            </div>
+          ))}
+        </div>
+      );
+    }
+    return (
+      <div style={styles.grid}>
+        {this.sortByDate(projects).map((project, index) => (
+          <div key={index} style={styles.card}>
+            <ProjectCard
+              projectData={project.projectData}
+              currentGallery={galleryType}
+            />
+          </div>
+        ))}
+      </div>
+    );
+  },
+
+  render() {
+
+    return  (
+      <div>
+        {this.renderSpecificGallery()}
```

Comment:
any reason not to just have all the code from `renderSpecificGallery` in your primary render function?